zeus/device/cpu/rapl.py

```diff
@@ -168,11 +166,14 @@ def cpus(self) -> Sequence[cpu_common.CPU]:
     def _init_cpus(self) -> None:
         """Initialize all Intel CPUs."""
         self._cpus = []
-        for index in range(2):
-            try:
-                self._cpus.append(RAPLCPU(index))
-            except ZeusRAPLFileInitError:
-                continue
+        pattern = re.compile(r"intel-rapl:(\d+)")
```

Comment:
Avoid regex for everything except when there's no good solution except for regex.

```python
from glob import glob

for dir in sorted(glob(f"{RAPL_DIR}/intel-rapl:*")):
    ...
```